### PR TITLE
pdnssec improvements

### PR DESCRIPTION
**Pull request by pieterlexis**
New features and improvements to pdnssec. Also adds a correct `wirelength` function to `DNSName`s.

@cmouse could you verify the PKCS#11 output of `list-keys`?

### CODE REVIEW

**Comment by cmouse** [comment]
no list of tsig keys?

**Comment by cmouse** [comment]
PKCS#11 code should report key <invalid> if it's size is 0.

**Comment by cmouse** [comment]
Won't work with ECDSAP256SHA256, ends up in infinite loop printing whitespace.

File: pdns/dnsname.cc [hunk 1]
```diff
@@ -106,6 +106,15 @@ size_t DNSName::length() const {
...
 }
 
+/**
+ * Get the length of the DNSName on the wire
+ *
+ * @return the total wirelength of the DNSName
+ */
+size_t DNSName::wirelength() const {
+  return d_storage.length() + 1;
+}
```
Comment (Habbie) [inline pdns/dnsname.cc]: a test for this would be neat

File: pdns/pdnssec.cc [hunk 2]
```diff
@@ -1406,12 +1604,25 @@ try
...
+        dbname = cmds[1];
+      } else {
+        if (cmds[1] != "overwrite"){
+          cerr << "Syntax: pdnssec create-bind-db [overwrite] FNAME"<<endl;
+          return 0;
+        }
+        overwrite = true;
+        dbname = cmds[2];
+      }
+      SSQLite3 db(dbname, true, overwrite); // create=ok
```
Comment (Habbie) [inline pdns/pdnssec.cc]: I forgot why we want the overwrite feature

File: pdns/ssqlite3.cc [hunk 3]
```diff
@@ -137,12 +137,17 @@ class SSQLite3Statement: public SSqlStatement
...
 {
-  // Open the database connection.
-  if(!creat) 
-    if ( access( database.c_str(), F_OK ) == -1 )
+  if (access( database.c_str(), F_OK ) == -1){
+    if (!creat)
       throw sPerrorException( "SQLite database '"+database+"' does not exist yet" );
+  } else {
+    if (creat && overwrite)
+      remove(database.c_str());
```
Comment (Habbie) [inline pdns/ssqlite3.cc]: `unlink` probably clearer here

File: pdns/pdnssec.cc (hunk 2 above)
Comment (pieterlexis) [inline pdns/pdnssec.cc]: Because we now warn that a database already exists, we should also be able to say 'just delete it'

Comment (Habbie) [inline pdns/pdnssec.cc]: I can't say I see that logic :)

**Comment by pieterlexis** [comment]
@Habbie we haz tests :smile:

**Pull request merged**